### ISSUE TITLE
Self-hosting firefly-iii and csv-importer on same server

### ISSUE THREAD

**Issue by 5833tP2** [firefly-iii/firefly-iii#3815]
I am running Firefly III version 5.3.3, self-hosted on a Pi 3. Total noob and out of my depth with running web applications ... migrating from Gnucash on a Linux desktop. I got firefly-iii working fine, once, but it all went bad when I tried to install and configure csv-importer. Now, nothing is working and I just get `404` errors in the browser. I'm probably missing something obvious but do not have the experience to work out what it might be.

**Description**
Pi is running on a local network only, accessible only with an IP number 192.168.1.45 (say).

LAMP stack installed with MySQL - all OK
I can enable/disable sites using `a2ensite` and [rewrite is enabled](https://docs.firefly-iii.org/faq/self_hosted#i-get-a-404)
Two sites are enabled `firefly-iii` and `csv-importer`

**firefly-iii.conf**
```
<VirtualHost *:80>
	ServerName 192.168.1.45/firefly-iii
	ServerAdmin webmaster@localhost
	DocumentRoot /var/www/firefly-iii/public

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>
```

**csv-importer.conf**
```
<VirtualHost *:80>
	ServerName 192.168.1.45/csv-importer
	ServerAdmin webmaster@localhost
	DocumentRoot /var/www/csv-importer/public

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>
```

I think that ServerName might be wrong in both files: the only online examples I could find were for sites with different domains but I've only got an IP number.

Regardless of what I type into the browser (`192.168.1.45` `192.168.1.45/firefly-iii` or `192.168.1.45/csv-importer` I just get a `404` error back.

It can't be rocket science to get these two applications running side by side, but so far it's eluded me. Can anyone ELI5 and help me work it out?

Thanks!

**Comment by JC5** [comment]
There's no need to indicate you need help. Very few people open issues just to socialize. Thanks.

Running two sites on one IP address can work but you'll have to fake your DNS entries. That's the easiest. What system are you running yourself? Windows, Mac?

**Comment by 5833tP2** [comment]
I'm running firefly-iii under Raspberry Pi OS / Debian 10.4 ... and it's just on a local network, not exposed to the Internet.

And thanks for adding the Question label to my post. I could work out how to do this myself. New to Github too.

**Comment by JC5** [comment]
No worries, I don't even think you could. But not the Raspberry but your own PC. Windows?

**Comment by 5833tP2** [comment]
Own PC is Manjaro Linux (5.8.6-1-MANJARO)

**Comment by JC5** [comment]
Cool, that makes it easier. This is the easiest solution because running Firefly III and/or the CSV importer in subdirectories is a nightmare.

Change the configs on the RPi to this:

```
<VirtualHost *:80>
	ServerName firefly-iii.home
	ServerAdmin webmaster@localhost
	DocumentRoot /var/www/firefly-iii/public

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>
```

```
<VirtualHost *:80>
	ServerName csv-importer.home
	ServerAdmin webmaster@localhost
	DocumentRoot /var/www/csv-importer/public

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>
```

Then, on both the Raspberry PI *and* oin your Manjaro PC, edit `/etc/hosts` and add the following lines:

```
# Firefly III 
192.168.1.45 firefly-iii.home
192.168.1.45 csv-importer.home
```

This will make them available from your home PC under http://firefly-iii.home and http://csv-importer.home

**Comment by 5833tP2** [comment]
Woohoo! That's great. I get to the `firefly-iii` login page fine, but have a Personal Access Token error on the `csv-importer` page. That's a different issue and I'll sort it out in due course.

So, the solution was to have a unique `ServerName` in each `.conf` file, and to make sure that both server and client `hosts` files contained the IP number and server names. Makes perfect sense.

Thanks so much for your help.
